lib/rex/parser/arguments.rb

```diff
@@ -154,6 +158,29 @@ def short_arg_from_long_arg(long_arg)
 
         fmt_option.each { |opt| return opt if opt =~ SHORT_FLAG }
       end
+
+      # Parsing takes into account longest `short` flag found in 'arg' should as extracted first
+      #
+      # Returns Array of short arguments found in `arg`
+      def short_args_from_token(arg)
+        compare_arg = arg.dup[1..-1]
+        short_args = []
+        found_args = {}
+        fmt.keys.each do |keys|
+          if keys.first =~ SHORT_FLAG
+            short_args << keys.first[1..-1]
+          end
+        end
+        short_args.sort_by! { |value| value.downcase }.reverse!
```

Comment:
May be confusing for some as it seems like this is actually editing the values in `short_args` but this is just downcasing so that everything is the same case, and then organizing things in reverse alphabetical order. It does not actually change the casing of `short_args` itself, it merely sorts it in reverse alphabetical order ignoring case differences.

Reply:
Yep, this is one of those, `if you know you know` Ruby things, since the concept it is used more than once in this file I left it uncommented. ¯\\_(ツ)_/¯